tt_eager/tt_dnn/op_library/reduce/reduce_op.cpp

```diff
@@ -210,10 +220,75 @@ Tensor mean(const Tensor& input_tensor,uint aggregate_dims /* = 2 */, const Memo
+    }
+
+    // Other sum dims will autoformat first before doing composite operations
+    Device * device;
+
+    // Get the device
+    if (input_tensor.storage_type() == StorageType::OWNED) {
+        device = AutoFormat::GetDefaultDevice();
+        TT_ASSERT(device != nullptr, "Requires setting default device if no inputs to op are on device");
+    } else {
+        device = input_tensor.device();
+    }
+
+    // @tt-aho TODO: Profile/determine which is more performant
+    // reduce sum on h
+    // 1 extra transpose wh and reduce sum on w
+    // Fused tranpose cw and reduce sum on w
+    if ( dim == 1 ) {
+        // Pad before running the op to only pay cost of formatting once
+        auto input_tensor_pad_shape = AutoFormat::pad_to_tile_shape(input_tensor.shape(), true);
+        auto out_shape = input_tensor.shape();
+        out_shape[1] = 1;
+
+        auto formatted_input_tensor = input_tensor;
+        if (!AutoFormat::check_input_tensor_format(input_tensor, input_tensor_pad_shape)) {
+            formatted_input_tensor = AutoFormat::format_input_tensor(input_tensor, device, input_tensor_pad_shape, 0.0, Layout::TILE);
+        }
+        Tensor output = transpose(formatted_input_tensor, 1, -2, output_mem_config);
+        output = reduce_impl<operatorValue>(output, 2, output_mem_config);
+        output = transpose(output, 1, -2, output_mem_config);
+        return AutoFormat::format_output_tensor(output, out_shape, device, Layout::TILE);
+    } else {
+        // Pad before running the op to only pay cost of formatting once
+        auto input_tensor_pad_shape = AutoFormat::pad_to_tile_shape(input_tensor.shape(), false, true);
+        auto out_shape = input_tensor.shape();
+        out_shape[0] = 1;
+
+        auto formatted_input_tensor = input_tensor;
+        if (!AutoFormat::check_input_tensor_format(input_tensor, input_tensor_pad_shape)) {
+            formatted_input_tensor = AutoFormat::format_input_tensor(input_tensor, device, input_tensor_pad_shape, 0.0, Layout::TILE);
+        }
+        Tensor output = transpose(input_tensor, 0, -2, output_mem_config);
+        output = reduce_impl<operatorValue>(output, 2, output_mem_config);
+        output = transpose(output, 0, -2, output_mem_config);
+        return AutoFormat::format_output_tensor(output, out_shape, device, Layout::TILE);
+    }
+}
+
+Tensor max(const Tensor &input_tensor, uint dim, const MemoryConfig& output_mem_config) {
+    return reduce_impl<ReduceOpMath::MAX>(input_tensor, dim,output_mem_config);
+}
+
 Tensor sum(const Tensor &input_tensor, uint dim, const MemoryConfig& output_mem_config) {
     TT_ASSERT( dim >= 0 && dim <= 3, "dimension have to be 0-3 only corresponding to N,C,H,W");
     constexpr float scaler1 = 1.0;
 
+    if (is_arch_whb0(input_tensor.device()->arch())) {
+      return reduce_impl<ReduceOpMath::SUM>(input_tensor, dim,output_mem_config);
+    }
+
```

Comment:
Why can't this just call `reduce_impl<ReduceOpMath::SUM>` like in the case for max?

What is the difference?